File: frontend-app/src/model/dto/response/DispositivoResponse.ts
import type { EmpresaResponse } from "./EmpresaResponse";

export type DispositivoStatus = "ATIVO" | "INATIVO";

export interface DispositivoResponse {
    id : number;
    host : string;
    nome : string;
    descricao : string;
    localizacao : string;
    sendoMonitorado : boolean;

    status : DispositivoStatus;
    empresa : EmpresaResponse;
}

File: main-api/src/main/java/com/redemonitor/main/dto/response/DispositivoResponse.java
package com.redemonitor.main.dto.response;

import com.redemonitor.main.enums.DispositivoStatus;

import lombok.*;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class DispositivoResponse {

    private Long id;
    private String host;
    private String nome;
    private String descricao;
    private String localizacao;
    private boolean sendoMonitorado;
    private DispositivoStatus status;
    private EmpresaResponse empresa;

}
